### PR TITLE
Refactor `TransactionDesc::FormatTxStatus` and ` TransactionStatus`

### PR DESCRIPTION
**Pull request by w0xlt**
This PR implements the changes suggested in https://github.com/bitcoin-core/gui/issues/538#issuecomment-1021913294 .

. remove redundant scope, rename `nDepth` -> `depth`, remove unused parameters and add translator comments in `TransactionDesc::FormatTxStatus()`
.  Use member initializers and remove unused field `qint64 TransactionStatus::open_for` in `TransactionStatus`.

Closes https://github.com/bitcoin-core/gui/issues/538

### CODE REVIEW

File: src/qt/transactiondesc.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2011-2021 The Bitcoin Core developers
+// Copyright (c) 2011-2022 The Bitcoin Core developers
```
Comment (jonatack) [inline src/qt/transactiondesc.cpp:1]: 38316d4 Generally there's no need to touch the copyright headers; they are updated annually with a script.

**Review by jonatack — COMMENTED** [review]
Code review ACK cbf23e493c7b448734119818217cc41865524278 modulo a couple comments

File: src/qt/transactiondesc.cpp (hunk 1 above)
Comment (w0xlt) [reply src/qt/transactiondesc.cpp]: Thanks. Done.

File: src/qt/transactionrecord.h [hunk 2]
```diff
@@ -40,22 +35,19 @@ class TransactionStatus
     };
 
     /// Transaction counts towards available balance
-    bool countsForBalance;
+    bool countsForBalance{false};
     /// Sorting key based on status
-    std::string sortKey;
+    std::string sortKey{};
```
[the remark this answers is not in the document]
Comment (w0xlt) [reply src/qt/transactionrecord.h]: Thanks. Done.

File: src/qt/transactionrecord.h [hunk 3]
```diff
@@ -20,12 +20,7 @@ struct WalletTxStatus;
 
 /** UI model for transaction status. The transaction status is the part of a transaction that will change over time.
  */
-class TransactionStatus
-{
-public:
-    TransactionStatus() : countsForBalance(false), sortKey(""),
-                          matures_in(0), status(Unconfirmed), depth(0), open_for(0)
-    { }
+struct TransactionStatus {
```
Comment (jonatack) [inline src/qt/transactionrecord.h:24]: b0ffe4e47c0500bcd54f82e6b88a0941ca nit if you retouch, can drop the extra newline

**Review by jonatack — COMMENTED** [review]
Code review ACK b0ffe4e47c0500bcd54f82e6b88a0941cae9485e

**Review by hebasto — COMMENTED** [review]
Approach ACK 0b234f96eaf85bcbe106e12750eefc57e009170d.

In 0b234f96eaf85bcbe106e12750eefc57e009170d "qt: remove redundant scope":
- it would be nice to rename s/`nDepth`/`depth`/
- parameters `const interfaces::WalletTx& wtx` and `int numBlocks` are not used, and could be dropped (maybe in a separated commit)
- if you feel comfortable doing that, translator comments are always welcome :)

In b0ffe4e47c0500bcd54f82e6b88a0941cae9485e "qt: refactor TransactionStatus":
- removing of `open_for` member could be split into a separated commit
- `bool needsUpdate` is still uninitialized
- maybe put into commit message something like this " qt, refactor: Use member initializers in TransactionStatus"?

Also could you please, in addition to the link to #538, add actual descriptions of this PR changes to OP, because it goes into a merge commit message.

File: src/qt/transactionrecord.h (hunk 3 above)
Comment (hebasto) [reply src/qt/transactionrecord.h:24]: Applying [`clang-format-diff.py`](https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy) after each commit is a good habit :)

Comment (w0xlt) [reply src/qt/transactionrecord.h]: Done. Thanks.

**Comment by w0xlt** [comment]
@hebasto I have addressed all of your suggestions. Thanks.

File: src/qt/transactionrecord.h [hunk 4]
```diff
@@ -40,19 +34,19 @@ class TransactionStatus
...
+    int matures_in{0};
     /**@}*/
 
     /** @name Reported status
        @{*/
-    Status status;
-    qint64 depth;
+    Status status{Unconfirmed};
+    qint64 depth{0};
     qint64 open_for; /**< Timestamp if status==OpenUntilDate, otherwise number
```
Comment (promag) [inline src/qt/transactionrecord.h]: b279331569e42cab721949d6669a452d6ba1558a

nit, initialize `open_for` or move this commit after 064bbdf243?

**Review by promag — APPROVED** [review]
Code review ACK b279331569e42cab721949d6669a452d6ba1558a

File: src/qt/transactiondesc.cpp [hunk 5]
```diff
@@ -36,13 +36,17 @@ QString TransactionDesc::FormatTxStatus(const interfaces::WalletTxStatus& status
 {
     int depth = status.depth_in_main_chain;
     if (depth < 0) {
+        //: This status represents an unconfirmed transaction that conflicts with another
         return tr("conflicted with a transaction with %1 confirmations").arg(-depth);
     } else if (depth == 0) {
         const QString abandoned{status.is_abandoned ? QLatin1String(", ") + tr("abandoned") : QString()};
+        //: This status represents an unconfirmed transaction that is in the memory pool or possibly abandoned
         return tr("0/unconfirmed, %1").arg(inMempool ? tr("in memory pool") : tr("not in memory pool")) + abandoned;
```
Comment (jarolrod) [inline src/qt/transactiondesc.cpp:44]: A translator comments applied to a line with multiple independent translatable strings will only apply to the first one. You can check what translator comment is applying to what string by making the translations, see more here: [translation_process.md](https://github.com/bitcoin/bitcoin/blob/master/doc/translation_process.md)

You could break this into multiple lines in order to give translator comments to the strings `in memory pool` and `not in memory pool`. Seems useful to provide context for this strings.

**Review by jarolrod — COMMENTED** [review]
concept ACK

File: src/qt/transactionrecord.h (hunk 4 above)
Comment (w0xlt) [reply src/qt/transactionrecord.h]: Done. Thanks.

File: src/qt/transactiondesc.cpp (hunk 5 above)
Comment (w0xlt) [reply src/qt/transactiondesc.cpp]: I think I addressed your suggestion in 62ab92b.

**Review by Gemk83 — APPROVED** [review]

**Review by Gemk83 — APPROVED** [review]

**Comment by hebasto** [comment]
Almost ACK 2c5dbf2fc4556ff83009eca2178f860efcb73f8e except for 62ab92b5e992d9c5bbe251632acbc7a5df517510 "qt, refactor: add translator comments in `TransactionDesc::FormatTxStatus()`".

If we are adding translator comments, we should also strive to prevent splitting of strings into less meaningful parts.

Suggesting to separate that commit into a dedicated PR, as all of other commits are good to be merged.

---

@w0xlt Please name this PR in a bit meaningful way because the PR title goes into the project commit history.

**Comment by w0xlt** [comment]
@hebasto the mentioned commit was moved to PR https://github.com/bitcoin-core/gui/pull/583.

File: src/qt/transactiondesc.cpp (hunk 5 above)
Comment (w0xlt) [reply src/qt/transactiondesc.cpp:42]: Moved this commit to PR https://github.com/bitcoin-core/gui/pull/583.

**Review by hebasto — APPROVED** [review]
ACK 343f83d0886ae39c9dacb29762ce712711b2bad2, I have reviewed the code and it looks OK, I agree it can be merged.

**Comment by hebasto** [comment]
@jonatack @jarolrod @promag 

Mind having another (final?) look?

**Review by jarolrod — APPROVED** [review]
Code Review ACK https://github.com/bitcoin-core/gui/commit/343f83d0886ae39c9dacb29762ce712711b2bad2

Thanks for working on this refactoring. I have reviewed the code and agree that it can be merged.

**Pull request merged**